kickstart/phpspec - Running/Passing tests for Todo\Models

erikthedeveloper opened the pull request:
@kchunterdeluxe when you get a chance, maybe you can check out what I've got so far. You'll see some starter files I needed to create for some of the other classes.

Also, I created a BaseModel to extend rather than requiring `Eloquent` in each of our models.

Let's hold off on merging this until we get a few more commits in here.

We can push to this branch to add more commits after review/etc...

erikthedeveloper committed:
Updates composer.json (PHPSpec/PSR-4) adds phpspec.yml

See Issue #4
- Adds PSR-4 `"Todo\\": "app/Todo"`
- Adds phpspec/phpspec
- Adds benconstable/phpspec-laravel

erikthedeveloper committed:
Adds Todo\DummyClasses\Calculator spec, class, implemented method, and passing test

erikthedeveloper committed:
See #3 - Adds beginnings of spec and Model for Task

- Tests failing

`Call to a member function connection() on a non-object in /Users/eaybar/Projects/GitHub_Repos_Misc/erikthedeveloper/todo-laravel-angularjs-homestead/TheApp/LaravelApi/vendor/laravel/framework/src/Illuminate/Database/Eloquent/Model.php on line 2782`

erikthedeveloper committed:
See #3 - Adds BaseModel to extend Todo\Models from, Task, and TaskSpec

- Adds `app/Todo/Models/BaseModel.php`
- Spec/Define relationships
    - Task belongsTo('Todo\Models\User');
    - Task hasMany('Todo\Models\Tag');
    - Task hasMany('Todo\Models\Comment');

erikthedeveloper committed:
See #3 - Updates testing.database config (copies development.database)

erikthedeveloper committed:
See #3 - Adds Comment, Tag, and User starter Classes

- Only declares namespace and extends BaseModel

@kchunterdeluxe I needed to add these to get the tests passing ... I wonder if there is a better way around that.
These files should get you started. I'm sticking with just the Task and Comment classes.

Commit:
See #3 - Adds spec tests for User and Tag classes

kaseyjcowley commented:
@erikthedeveloper 
OK, so I need to learn how to use github better. I pushed up my changes, but I should have pushed to a different branch and submitted a pull request. Live and learn. You can check out what I did and we'll move to the next phase. I think the relationship tests I wrote are correct for our data model.

erikthedeveloper committed:
Removes DummyClasses directory in both spec and Todo

erikthedeveloper committed:
Declares methods as public

erikthedeveloper committed:
Adds and modifies spec/model for Comment

- should belong to one user

erikthedeveloper committed:
Changes testing database to in memory sqlite

erikthedeveloper committed:
Changes Tag spec/model - Updates relationships defined

- Tags have a many-to-many relationship with Tasks
- A Tag does not belong, directly, to a User

@kcunterdeluxe I've got some migrations/tests I'm working on to show this.

erikthedeveloper committed:
Adds 3 migrations - New tables: tags, tasks, and tag_task (rough draft)

- Unmodified since `generate:migration...`
- Needs work

@kchunterdeluxe Let's try and keep one migration per table. We can continually:
    - Edit the migration files
    - Run `migrate:refresh`

erikthedeveloper committed:
Adds migrate_db to phpspec.yml
